src/node_messaging.cc

```diff
@@ -447,18 +447,18 @@ MessagePort::MessagePort(Environment* env,
                          Local<Object> wrap)
   : HandleWrap(env,
                wrap,
-               reinterpret_cast<uv_handle_t*>(new uv_async_t()),
+               reinterpret_cast<uv_handle_t*>(&async_),
                AsyncWrap::PROVIDER_MESSAGEPORT),
     data_(new MessagePortData(this)) {
   auto onmessage = [](uv_async_t* handle) {
     // Called when data has been put into the queue.
-    MessagePort* channel = static_cast<MessagePort*>(handle->data);
+    MessagePort* channel = ContainerOf(&MessagePort::async_, handle);
```

Comment:
I wanted to ask this earlier as well - what is the consideration behind obtaining `this` through pointer arithmetic over the other one?